w04-associations-and-auth/project-1/README.md

```diff
@@ -0,0 +1,102 @@
+* **User stories.** What features will your app have? Set up your project and user stories in [Trello](https://trello.com).
+* **Wireframes.** Sketch out what each of your pages will look like and how they will work. Consider making a _paper prototype_ to demonstrate and/or test key user interactions.
+* **Object Models.** Draw out the models and any associations for your project in an entity relationship diagram (ERD).
+* **Milestones.** What are major steps to completing the project?
+* **Feasibility Check.** If you are using an external API, make sure you can get that data. If you are using a new tool, go through its getting started tutorial. We will ask to see your results.
+
+
+### Completed project deliverables:
+
+* Link to Heroku hosted project, with core and 3 flex technical requirements complete
+* Link to source code on GitHub
+* A `README` with the following:
+  * `Description`: A paragraph-long elevator pitch of your project
+  * Another Link to project hosted on Heroku
+  * Wireframes
+  * `User Stories`, `Site Flow`, or equivalent
+  * An Entity Relationship Diagram
+  * APIs used
+  * Libraries used
+  * Wishlist / Future Development
+
+## TECHNICAL REQUIREMENTS
+
+### Core Technical Requirements
+
+**Your app should have all of the following:**
+
+* **Express API** Implement a server-side JSON API with Express.
+* **RESTful Routes** Design the routes in a <a href="http://restfulrouting.com/mappings/resources" target="_blank">RESTful</a> manner.
+* **MongoDB** Persist data with at least two models in a Mongo Database.
+* **AJAX** Leverage your server-side Express API to fetch JSON data asynchronously to the client-side.
+* **jQuery** Use jQuery to manipulate the DOM and listen for events on the client-side.
+* **Server-Side HTML Templating** Render your HTML views with data using EJS templating.
+* **Cookies or Sessions** Save some data in cookies or sessions.
+* **Visual Design** Leverage Bootstrap to kick-start your UI and UX.
+* **Custom Styling** Go beyond Bootstrap with styles and/or animations customized for your app. 
+* **Heroku** Deploy your app to Heroku.  Ensure no app secrets are exposed. __Do not commit secret keys to GitHub!__
+
+### Flex Technical Requirements
+
+**Your app should have 3 of the 4 following options:**
+* **External API** Use an external API to integrate rich data into your app.
+* **Model Relationship** Create a one-to-many or many-to-many relationship between two models using either embedded or referenced data.
+* **Authentication** Enable users to sign up, log in, and log out. (Note: This covers the core sessions or cookies requirement.)
+* **Data Validation** On the client side, don't let users submit blank forms -- and give them a visible error message explaining why they weren't allowed to submit. In your database, use mongoose's validation for at least one attribute for each of your schemas.

```

Comment:
* should it be "mongoose's validations"
* this is untaught, should we link to http://mongoosejs.com/docs/validation.html or a better tutorial